### PR TITLE
fix(nfc): Report card error type on card errors during reconstruction

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/PRF-6169

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6169 Device is sending "unknown X1 card error" on cysync when wallet is locked](https://app.clickup.com/t/9002019994/PRF-6169)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6169 Device is sending "unknown X1 card error" on cysync when wallet is locked](https://app.clickup.com/t/9002019994/PRF-6169)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6169 Device is sending "unknown X1 card error" on cysync when wallet is locked](https://app.clickup.com/t/9002019994/PRF-6169)

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
change according to review comments.

File: src/wallet/reconstruct_wallet_flow.c [hunk 1]
```diff
@@ -256,6 +265,8 @@ static reconstruct_state_e reconstruct_wallet_handler(reconstruct_state_e state,
          * These error codes are non-recoverable from this flow. Error message
          * will be displayed to the user before the main menu
          */
+        reject_cb(ERROR_COMMON_ERROR_CARD_ERROR_TAG,
+                  get_card_error_from_nfc_status(card_error_code));
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/wallet/reconstruct_wallet_flow.c:269]: Will update for CARD_OPERATION_P0_OCCURED.

Here CARD_OPERATION_ABORT_OPERATION is not P0 abort.
It indicates that the error is non-recoverable for the user. Refer enum comments

[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/wallet/reconstruct_wallet_flow.c:271]: updated re-review

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**